base-local/docker-compose.yaml.fs

```diff
@@ -1,103 +1,97 @@
-version: '3.6'
-services:
+version: "3.6"
 
-    database:
-        image: "mariadb:latest"
-        ports:
-          - "3326:3306"
-        restart: always
-        environment:
-            MYSQL_ROOT_PASSWORD: "secretpassword"
-        volumes:
-          - "dbdata:/var/lib/mysql"
+x-app: &default-app
+  environment:
+    - "READIUM_DATABASE_HOST=${READIUM_DATABASE_HOST}"
+    - "READIUM_DATABASE_PORT=${READIUM_DATABASE_PORT}"
+    - "READIUM_DATABASE_USERNAME=${READIUM_DATABASE_USERNAME}"
+    - "READIUM_DATABASE_PASSWORD=${READIUM_DATABASE_PASSWORD}"
+    - "READIUM_LCPSERVER_HOST=http://${READIUM_LCPSERVER_HOST}:${READIUM_LCPSERVER_PORT}"
+    - "READIUM_LCPSERVER_PORT=${READIUM_LCPSERVER_PORT}"
+    - "READIUM_LCPSERVER_DATABASE=${READIUM_LCPSERVER_DATABASE}"
+    - "READIUM_LCPSERVER_USERNAME=${READIUM_LCPSERVER_USERNAME}"
+    - "READIUM_LCPSERVER_PASSWORD=${READIUM_LCPSERVER_PASSWORD}"
+    - "READIUM_LSDSERVER_HOST=http://${READIUM_LSDSERVER_HOST}:${READIUM_LSDSERVER_PORT}"
+    - "READIUM_LSDSERVER_PORT=${READIUM_LSDSERVER_PORT}"
+    - "READIUM_LSDSERVER_DATABASE=${READIUM_LSDSERVER_DATABASE}"
+    - "READIUM_FRONTEND_HOST=http://${READIUM_FRONTEND_HOST}:${READIUM_FRONTEND_PORT}"
+    - "READIUM_FRONTEND_PORT=${READIUM_FRONTEND_PORT}"
+    - "READIUM_FRONTEND_DATABASE=${READIUM_FRONTEND_DATABASE}"
+    - "READIUM_ENC_CONTENT=/opt/readium/files/encrypted"
```

Comment:
In this simple use case (one's own local computer), this "every variable for every service" gets sent to every container (if I'm reading it correctly) is okay. It kind of enables a DRY-style reuse. But in some other use case, it'd be better to reference the required variables only in each container to avoid over-exposing secrets.

Reply:
I'm concerned about this too. The problem is that all of those variables are used in `etc/config.yaml` and have to be present in the container for `envsubst` to work correctly. The solution would be to split the configuration file

Reply:
Sounds good. We'll leave it for now. If/when I get to a multi-host deployment, we will split up the config file and do something different.